src/crewai/utilities/embedding_configurator.py

```diff
@@ -91,14 +91,51 @@ def _configure_azure(config, model_name):
             organization_id=config.get("organization_id"),
         )
 
+    @staticmethod
+    def _validate_url(url):
+        """Validate that a URL is properly formatted and uses HTTP/HTTPS scheme."""
+        if not url:
+            return False
+        
+        from urllib.parse import urlparse
+        
+        try:
+            result = urlparse(url)
+            return all([
+                result.scheme in ('http', 'https'),
+                result.netloc
+            ])
+        except ValueError:
+            return False
+
+    @staticmethod
+    def _construct_embeddings_url(base_url):
+        """Construct the full embeddings URL from a base URL."""
+        if not base_url:
+            return "http://localhost:11434/api/embeddings"
+        
+        base_url = base_url.rstrip('/')
+        return f"{base_url}/api/embeddings" if not base_url.endswith('/api/embeddings') else base_url
```

Comment:
you are assuming that all embeddings providers's url has the same format. It might not be totally truth. 

I think we don't need this method, please drop it and then use the provided byt config or env var